Source/WTF/wtf/NativePromise.h

```diff
@@ -1192,42 +1229,54 @@ class NativePromise final : public NativePromiseBase, public ConvertibleToNative
         Storage()
             : m_result([] {
                 if constexpr(IsExclusive)
-                    return std::nullopt;
+                    return NoResult { };
                 else
                     return adoptRef(*new RefCountedResult);
             }())
         {
         }
         bool has_value() const
         {
-            if constexpr (IsExclusive)
-                return m_result.has_value();
-            else
-                return m_result->result.has_value();
+            return std::holds_alternative<Result>(optionalResult());
```

Comment:
has_value() is a bit odd now since it does not account for the possiblity to get a result from the runnable.
This has an impact on `isResolved()` getter for instance.
I would think that a promise that has a runnable actually has a value.

Reply:
I have renamed the method to `hasResult()`